book/src/concepts/type-conversions.md

```diff
@@ -0,0 +1,60 @@
+# Type Conversions
+
+As the primary use for CXX-Qt is to describe a Qt API in Rust, this means that for technical reasons the required Qt API will sometimes be slightly different from the Rust API.
+
+For example, when dealing with [opaque types](./types.html#custom-opaque-types), these can never be passed by-value from/to Rust.
+In most cases, you'll therefore use a UniquePtr in Rust when you need ownership of such an object.
+
+However, Qt often requires values to be returned by-value, especially when compatibility with QML is required.
+CXX-Qt therefore allows you to automatically convert certain Rust types to certain C++ types.
+
+For example, CXX-Qt can convert a `std::unique_ptr<QColor>` to a `QColor` for you.
+
+CXX-Qt by default provides conversions for:
+
+| From                  | To            |                                           |
+|-----------------------|---------------|-------------------------------------------|
+| any type `T`          | any type `R`  | If `T` can be implicitly converted to `R` |
+| `std::unique_ptr<T>`  | T             | if T is moveable |
+| `std::unique_ptr<T>`  | T&            | |
+
+## Defining your own converter
+CXX-Qt uses the templated `rust::cxxqtlib1::cxx_qt_convert` struct for these conversions.
+In fact, an error message on this struct might have brought you here 😉.
+
+Before starting to define your own type converter, make sure this is actually what you want!!!
+Is the C++ compiler complaining about two types that you think conversion should be possible between?
+If not, you might have a bug in your `cxx_qt::bridge`.
+Double-check that you know where the error originates from and that the types are correct.
+
+However, if you're confident that there is a conversion missing, consider [opening an issues on the CXX-Qt repo](https://github.com/KDAB/cxx-qt/issues/new).
+Then you can use template specialization on the `cxx_qt_convert` struct to provide your own type conversion that CXX-Qt will use.
+
+Example:
+``` c++
```

Comment:
should this have `c++,ignore` ?

Reply:
I don't think that makes sense for c++, as the doc tests don't compile those anyway.